### DISCUSSION TITLE
Spotify is locking us out of our own accounts again

### DISCUSSION THREAD

**Discussion by rwjack** [librespot-org/librespot#1311] [General]
"To protect your privacy, we will be fucking you over and making you login on every device again"

**Comment by norio-sh** [comment]
So, whats the status on this issue? Does this happen to everybody? I have tried several times, but every time I get the message from Spotify and have to reset the password and log in on all my devices again, which is really a pain!!

**Reply by norio-sh** [reply]
I switched to token based authentication and that worked fine, no more account lockout

**Comment by DaXcess** [comment]
Started experiencing this recently as well, and receiving reports of more people having their passwords reset as well.
Due to the inconsistency of the resets it's kinda hard to tell what exactly is causing this.

 - It could be the version numbers librespot uses when connecting to the APs'
 - It could be the way authentication works (in my case) in 0.5.0-dev, where token auth gets upgraded to reusable credentials and re-authed
 - It could literally be anything else (Spirc? old Harmony stuff?)

I could try and experiment to see if using cached credentials more often might reduce these random password resets

**Reply by kingosticks** [reply]
> I could try and experiment to see if using cached credentials more often might reduce these random password resets

I don't understand. After the very first login, why wouldn't you **always** be using cached credentials? This is the only sensible way to configure things.

**Reply by DaXcess** [reply]
TLDR: Legacy code

I'm using librespot as a library, and I first perform a "are these credentials still valid yet?" sanity check before actually invoking librespot, which causes either one of the following to happen:

- No known cache: token auth -> grab reusable creds -> invoke librespot
- Known cache: check if valid, disconnect, and then invoke librespot, causing a login to happen twice

I use this system since librespot didn't re-auth token authentication when I started using it. Now that librespot *does* do this I can rewrite that mess to connect instantaneously, and only re-auth if the stored creds have expired

**Reply by kingosticks** [reply]
I'm curious, in what situation do stored creds expire for you?

**Reply by DaXcess** [reply]
They "expire" after password resets. In any case, it's something that can fail, so I (currently) check it to see if it's still valid.

**Comment by kingosticks** [comment]
@roderickvd did you hear anything back from that reply to Spotify when they reached out about deprecating username/password login? I'd like to know what their rules are regarding account resets, do you think they'd help?

**Reply by roderickvd** [reply]
Nothing, unfortunately. Silence.
